Content.Client/Power/SolarPanelComponent.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Content.Client.Power
+{
+    /// <summary>
+    ///     This is a solar panel.
+    ///     It generates power from the sun based on coverage.
+    /// </summary>
+    [RegisterComponent]
+    public sealed class SolarPanelComponent : Component
```

Comment:
if the SolarPanelComponent is defined in the server, just move the definition to shared and add the  [NetworkedComponent] attribute rather than making an empty definition over here.

Reply:
Done. Doesn't this, however, allocate the component variables also on the client? Now that I think about it, I could've made the shared component empty and the server component with the properties; is it a good practice to have a networked component just for the clients and then a specific component for the server?

Reply:
best practice is just to have a single component in shared. If needed, you can split it into server/client and use componentreference